packages/api/src/routes/medical/patient.ts

```diff
@@ -223,4 +229,58 @@ router.get(
   })
 );
 
+/** ---------------------------------------------------------------------------
+ * POST /patient/:id/consolidated
+ *
+ * Returns a Bundle with the outcome of the query.
+ *
+ * @param req.cxId The customer ID.
+ * @param req.param.id The ID of the patient to associate resources to.
+ * @param req.body The FHIR Bundle to create resources.
+ * @return FHIR Bundle with operation outcome.
+ */
+router.post(
+  "/:id/consolidated",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getCxIdOrFail(req);
+    const patientId = getFrom("params").orFail("id", req);
+    const fhirBundle = bundleSchema.parse(req.body);
+    const patient = await getPatientOrFail({ id: patientId, cxId });
+    const validatedBundle = validateFhirEntries(fhirBundle);
+
+    if (Config.isSandbox()) {
+      const data = await getConsolidatedPatientData({ cxId, patientId });
+      // limit the amount of resources stored in sandbox mode
+      if (data.entry && data.entry.length >= MAX_RESOURCE_STORED_LIMIT) {
+        return res.status(status.BAD_REQUEST).json({
+          message: `Cannot create bundle with size greater than ${MAX_RESOURCE_STORED_LIMIT} bytes in Sandbox mode!`,
+        });
+      }
+    }
+
+    // Limit the amount of resources that can be created at once
+    if (validatedBundle.entry.length >= MAX_RESOURCE_POST_LIMIT) {
+      return res.status(status.BAD_REQUEST).json({
+        message: `Cannot create more than ${MAX_RESOURCE_POST_LIMIT} resources at a time!`,
+      });
+    }
+
+    const contentLength = req.headers["content-length"];
+    // limit the payload size that can be created
+    if (contentLength && parseInt(contentLength) >= MAX_CONTENT_LENGTH) {
+      return res.status(status.BAD_REQUEST).json({
+        message: `Cannot create bundle with size greater than ${MAX_CONTENT_LENGTH} bytes!`,
```

Comment:
nit: exclamation point doesn't add to the context/message, and can be taken as aggressive. Not a huge deal, but I'd avoid using it.